### ISSUE TITLE
Salt: local minion (master on the same host) won't accept the RSA master PK

### ISSUE THREAD

**Issue by PierreR** [NixOS/nixpkgs#25434]
## Issue description
On a local nixos machine, running both a salt-master and a salt-minion as root, as soon as the master accepts the minion key, the minion won't validate the master key.

The minion crashes with:
```
  File "/nix/store/xrsm7m01pba4fp8xxrz485cdb2l8208s-salt-2016.11.2/lib/python2.7/site-packages/salt/utils/rsax931.py", line 139, in __init__
    raise ValueError('invalid RSA public key')
ValueError: invalid RSA public key
```

### Steps to reproduce

- add `salt` in `environment.systemPackages`
- start the salt master as root with: salt-master 
- start salt-minion as root with: salt-minion -l debug
- accept the minion key with salt-key -a

## Technical details

* System: 17.03.1057.c545296adc (Gorilla)
* Nix version: nix-env (Nix) 1.11.8

**Comment by aneeshusa** [comment]
cc me as maintainer
I believe I've run in the similar/the same issue, and last I checked I tracked it down to some issue with the dynamic loading of OpenSSL in `salt/utils/rsax931.py`. Unfortunately I wasn't able to debug it any further.

**Comment by aneeshusa** [comment]
I just added a Salt master module for NixOS in #25632, and I am able to have NixOS Salt minions successfully connect to my NixOS Salt master. That PR also includes a Salt version bump. I don't know which of these fixed the problem, but this is now working for me; I'll try to add a Salt minion service shortly.

**Comment by PierreR** [comment]
@aneeshusa Thanks. I will give it a shot shortly. 

As a note, the `2016-11.4 ` is a buggy release (https://github.com/saltstack/salt/issues/40845). `2016-11.5` should be released in a day or two.

**Comment by Mic92** [comment]
Please feel free to update salt and make a pull request, when that happens.

**Comment by aneeshusa** [comment]
FYI, 2016.11.5 was released today so I made #25910 to update it, already tested it locally.
